packages/api/src/beacon/client/beacon.ts

```diff
@@ -1,13 +1,46 @@
 import {ChainForkConfig} from "@lodestar/config";
-import {Api, ReqTypes, routesData, getReqSerializers, getReturnTypes} from "../routes/beacon/index.js";
-import {IHttpClient, generateGenericJsonClient} from "../../utils/client/index.js";
+import {Api, ReqTypes, routesData, getReqSerializers, getReturnTypes, BlockId} from "../routes/beacon/index.js";
+import {IHttpClient, generateGenericJsonClient, getFetchOptsSerializers} from "../../utils/client/index.js";
+import {ResponseFormat} from "../../interfaces.js";
+import {BlockResponse, BlockV2Response} from "../routes/beacon/block.js";
 
 /**
  * REST HTTP client for beacon routes
  */
 export function getClient(config: ChainForkConfig, httpClient: IHttpClient): Api {
   const reqSerializers = getReqSerializers(config);
   const returnTypes = getReturnTypes();
-  // All routes return JSON, use a client auto-generator
-  return generateGenericJsonClient<Api, ReqTypes>(routesData, reqSerializers, returnTypes, httpClient);
+  // Some routes return JSON, use a client auto-generator
+  const client = generateGenericJsonClient<Api, ReqTypes>(routesData, reqSerializers, returnTypes, httpClient);
+  const fetchOptsSerializer = getFetchOptsSerializers<Api, ReqTypes>(routesData, reqSerializers);
+
+  return {
+    ...client,
+    async getBlock<T extends ResponseFormat = "json">(blockId: BlockId, format?: T) {
+      if (format === "ssz") {
+        const res = await httpClient.arrayBuffer({
```

Comment:
This is sweet.  I didnt know our httpClient did so much under the hood.  was looking at how its implemented.